app/src/main/java/ru/practicum/android/diploma/ui/screen/FilterCountryScreen.kt

```diff
@@ -29,11 +56,81 @@ fun FilterCountryScreen(modifier: Modifier, onBackClick: () -> Unit) {
             modifier = modifier
                 .padding(padding)
                 .padding(horizontal = paddingBase),
-            verticalArrangement = Arrangement.Center,
             horizontalAlignment = Alignment.CenterHorizontally
         ) {
-            Button(onBackClick) {
-                Text(stringResource(R.string.filter_choose_label))
+            val state = viewModel.screenState.collectAsState().value
+            when (state) {
```

Comment:
Со стейтом лучше работать следующим образом:
```suggestion
            val uiState by viewModel.screenState.collectAsStateWithLifecycle()
            when (val state = uiState) {
```
Тогда вы явно не обращаетесь к `value` полю и используете более надёжный способ коллекта стейта - `collectStateWithLifecycle`